Sources/MapboxCoreNavigation/NavigationService.swift

```diff
@@ -491,13 +493,20 @@ public class MapboxNavigationService: NSObject, NavigationService {
         guard let routeResponse = history.initialRoute else {
             return nil
         }
+        
+        let replayLocationManager = ReplayLocationManager(history: history,
+                                                          listener: customHistoryEventsListener)
         self.init(indexedRouteResponse: routeResponse,
                   customRoutingProvider: customRoutingProvider,
                   credentials: credentials,
-                  locationSource: ReplayLocationManager(history: history),
+                  locationSource: replayLocationManager,
                   eventsManagerType: eventsManagerType,
                   routerType: routerType,
                   customActivityType: customActivityType)
+        
+        if customHistoryEventsListener == nil {
```

Comment:
I don't see use cases where this listener should be changed during a replay, so decided not to expose extra entities. Do you have any ideas why it could be needed?